packages/loot-core/src/platform/server/fs/index.electron.ts

```diff
@@ -111,11 +111,35 @@ export const readFile: T.ReadFile = (
 };
 
 export const writeFile: T.WriteFile = (filepath, contents) => {
-  return new Promise(function (resolve, reject) {
-    // @ts-expect-error contents type needs refining
-    fs.writeFile(filepath, contents, 'utf8', function (err) {
-      return err ? reject(err) : resolve(undefined);
-    });
+  const NO_OF_RETRIES = 5;
+  const INITIAL_RETRY_DELAY = 200;
+  let retryDelay = INITIAL_RETRY_DELAY;
+
+  return new Promise((resolve, reject) => {
```

Comment:
I like the idea! But it could be simplified with `promise-retry`.

https://github.com/actualbudget/actual/blob/master/packages/desktop-client/src/components/util/LoadComponent.tsx#L45-L46

Reply:
Done. I've also increased the retry count and reduced the timeout between retries. I think file locks are short lived so this should recover faster.